### PR TITLE
[CHORE] enable rule eslint-plugin/require-meta-type

### PR DESCRIPTION
**Pull request by armano2**
This PR enables `eslint-plugin/require-meta-type` rule, and adds type field to all existing rules.

fixes: #216

### CODE REVIEW

**Comment by j-f1** [comment]
Is there a list of the valid types and their meanings somewhere?

**Comment by armano2** [comment]
@j-f1 You can find them in eslint documentation https://eslint.org/docs/developer-guide/working-with-rules#rule-basics 

> type (string) indicates the type of rule, which is one of "problem", "suggestion", or "layout":
> - "problem" means the rule is identifying code that either will cause an error or may cause a confusing behavior. Developers should consider this a high priority to resolve.
> - "suggestion" means the rule is identifying something that could be done in a better way but no errors will occur if the code isn’t changed.
> - "layout" means the rule cares primarily about whitespace, semicolons, commas, and parentheses, all the parts of the program that determine how the code looks rather than how it executes. These rules work on parts of the code that aren’t specified in the AST.

**Comment by j-f1** [comment]
> i was thinking about generating additional config base on "layout" type (config-no-layout) to for prettier

I’m planning to submit a PR to [`eslint-config-prettier`](https://github.com/prettier/eslint-config-prettier) to add a `prettier/typescript` config to disable all formatting-related rules.

File: lib/rules/no-use-before-define.js [hunk 1]
```diff
@@ -180,6 +180,8 @@ function isInInitializer(variable, reference) {
 
 module.exports = {
     meta: {
+        type: "problem",
+
```
Comment (bradzacher) [inline lib/rules/no-use-before-define.js]: nit: newline

**Comment by armano2** [comment]
@bradzacher suggestions applied

**Pull request merged**